recipes/openscenegraph/all/conanfile.py

```diff
@@ -102,163 +104,254 @@ def config_options(self):
 
             # imageio supports tiff files so the tiff plugin isn't needed on Apple platforms
             self.options.with_tiff = False
+        else:
+            del self.options.with_avfoundation
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
         if not self.options.with_zlib:
             # These require zlib support
-            del self.options.with_openexr
-            del self.options.with_png
-            del self.options.with_dcmtk
+            self.options.rm_safe("with_openexr")
+            self.options.rm_safe("with_png")
+            self.options.rm_safe("with_dcmtk")
 
-    def validate(self):
-        if self.options.get_safe("with_asio", False):
-            raise ConanInvalidConfiguration("ASIO support in OSG is broken, see https://github.com/openscenegraph/OpenSceneGraph/issues/921")
-        if hasattr(self, "settings_build") and cross_building(self):
-            raise ConanInvalidConfiguration("openscenegraph recipe cannot be cross-built yet. Contributions are welcome.")
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        if self.options.enable_windowing_system and self.settings.os == "Linux":
+        if self.options.enable_windowing_system and self.settings.os in ["Linux", "FreeBSD"]:
             self.requires("xorg/system")
         self.requires("opengl/system")
 
         if self.options.use_fontconfig:
             self.requires("fontconfig/2.14.2")
 
-        if self.options.get_safe("with_asio", False):
+        if self.options.get_safe("with_asio"):
             # Should these be private requires?
-            self.requires("asio/1.22.1")
-            self.requires("boost/1.81.0")
+            self.requires("asio/1.28.2")
+            self.requires("boost/1.83.0")
         if self.options.with_curl:
-            self.requires("libcurl/8.0.1")
+            self.requires("libcurl/[>=7.78 <9]")
         if self.options.get_safe("with_dcmtk"):
-            self.requires("dcmtk/3.6.6")
+            self.requires("dcmtk/3.6.7")
         if self.options.with_freetype:
-            self.requires("freetype/2.13.0")
+            self.requires("freetype/2.13.2")
         if self.options.with_gdal:
-            self.requires("gdal/3.4.3")
+            self.requires("gdal/3.8.0")
```

Comment:
```suggestion
            self.requires("gdal/3.8.3")
```
this will reduce conflicts as well